h/api/models/document.py

```diff
@@ -26,17 +26,18 @@ class Document(Base, mixins.Timestamps):
     def __repr__(self):
         return '<Document %s>' % self.id
 
+    @property
+    def title(self):
+        meta = DocumentMeta.query.filter(DocumentMeta.document_id == self.id,
+                                         DocumentMeta.type == 'title').first()
+        if meta:
+            return meta.value
+
     @classmethod
-    def find_or_create_by_uris(cls, claimant_uri, uris,
```

Comment:
I'd suggest that given they're going to need access to the features fetcher, perhaps we should just add `request` to the start of every `storage` API function's argument list?